pkg/controller/reconciler/deployment_reconciler.go

```diff
@@ -248,13 +248,17 @@ func (r *DeploymentReconciler) reconcileUpdate(ctx context.Context, deployment *
 		}
 	}
 
+	if annotations.Configuration == nil {
+		annotations.Configuration = &deploymentConfiguration{}
+	}
```

Comment:
Similar to the last issue we looked at, I'd like to understand the root cause before we start making fixes. 

Do you know what set of conditions are causing this?

It should be possible to add a new integration test in `deployment_reconciler_test.go` that reproduces this issue.